src/globals/internal/FloatingMenu.js

```diff
@@ -78,24 +78,29 @@ const getFloatingPosition = ({
   const refCenterHorizontal = (refLeft + refRight) / 2;
   const refCenterVertical = (refTop + refBottom) / 2;
 
-  return {
-    [DIRECTION_LEFT]: () => ({
-      left: refLeft - width - left,
-      top: refCenterVertical - height / 2 + scrollY + top,
-    }),
-    [DIRECTION_TOP]: () => ({
-      left: refCenterHorizontal - width / 2 + left,
-      top: refTop - height + scrollY - top,
-    }),
-    [DIRECTION_RIGHT]: () => ({
-      left: refRight + left,
-      top: refCenterVertical - height / 2 + scrollY + top,
-    }),
-    [DIRECTION_BOTTOM]: () => ({
-      left: refCenterHorizontal - width / 2 + left,
-      top: refBottom + scrollY + top,
-    }),
-  }[direction]();
+  switch (direction) {
```

Comment:
I turned this into a switch statement because it seemed more readable, i didn't change any of the logic though

Reply:
👍

Reply:
now that I added `caretWidth` it is slightly different now

Reply:
Nice work. This is probably something we can contribute back upstream to carbon.